src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassWriter.java

```diff
@@ -845,7 +845,7 @@ void writeInnerClasses() {
                 !inner.name.isEmpty() ? poolWriter.putName(inner.name) : 0);
             databuf.appendChar(flags);
             icCount++;
-            if (inner.isPrimitiveClass()) {
+            if (inner.isPrimitiveClass() && types.splitPrimitiveClass) {
```

Comment:
the test `sym.isPrimitiveClass() && types.splitPrimitiveClass` is so common which I think would be better refactored into a separate predicate, to make the code clearer.

e.g. `Symbol.isSplitPrimitiveClass(Types)`

Reply:
Thanks, I will take a look